### PR TITLE
roll of dice implementation

### PR DESCRIPTION
**Pull request by briansetzesglobant**
Implementation of the roll of dice applying the bloc pattern

Gif of the application running
ezgif com-gif-maker

### CODE REVIEW

File: lib/src/bloc/dice_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import 'dart:async';
+import 'dart:math';
+import 'package:dice_roller/utils/constants.dart';
```
Comment (MarcosCGlobant) [inline lib/src/bloc/dice_bloc.dart]: missed a relative import here

File: lib/utils/constants.dart [hunk 2]
```diff
@@ -19,4 +19,8 @@ abstract class Constants {
   static const int textStyleGreen = 56;
   static const int textStyleBlue = 27;
   static const double textStyleOpacity = 2.0;
+
+  static const List<int> bodyList = [1, 2, 3, 4, 5, 6];
+  static const String body = "6";
```
Comment (MarcosCGlobant) [inline lib/utils/constants.dart]: You should use a more representative name for this constant, and change the number for the sum of your initial dices values, so you keep consistency on the initial screen

**Review by MarcosCGlobant — COMMENTED** [review]
Some minor changes needed

File: lib/src/bloc/dice_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  @override
+  Stream<String> get scoreStream => _scoreStreamController.stream;
+
+  @override
+  void diceRoll() {
+    _values.clear();
+    int sumOfTheScore = 0;
+    int diceValue = 0;
+    int i = 0;
```
Comment (FranciscoGaraventa) [inline lib/src/bloc/dice_bloc.dart]: Remember that although the method is public, the variables you use internally are private, therefore they should start with '_'

File: lib/src/bloc/dice_bloc.dart (hunk 1 above)
Comment (briansetzesglobant) [reply lib/src/bloc/dice_bloc.dart]: Done

File: lib/utils/constants.dart (hunk 2 above)
Comment (briansetzesglobant) [reply lib/utils/constants.dart]: Fixed

File: lib/src/bloc/dice_bloc.dart (hunk 3 above)
Comment (briansetzesglobant) [reply lib/src/bloc/dice_bloc.dart]: Fixed

**Review by FranciscoGaraventa — APPROVED** [review]
LGTM

**Review by MarcosCGlobant — APPROVED** [review]
LGTM

**Pull request merged**